__tests__/app.test.js

```diff
@@ -0,0 +1,56 @@
+const request = require('supertest');
+const {app, server} = require('../app.js');
+const db = require('../db/connection.js');
+const seed = require('../db/seeds/seed.js');
+const data = require('../db/data/test-data/index.js')
+
+beforeEach(() => {
+    return seed(data);
+});
+
+afterAll(() => {
+    return db.end().then(() => {return server.close()});
+})
+
+describe('Invalid endpoint requests', () => {
+    test('404: should return not found in response to get request to non-existent endpoint', () => {
+        return request(app)
+            .get('/api/does-not-exist')
+            .expect(404)
+    });    
+});
+describe('GET /api/topics', () => {
+    test('200: should return an array', () => {
+        return request(app)
+            .get('/api/topics')
+            .expect(200)
+            .then(({body}) => {
+                expect(Array.isArray(body.topics)).toBe(true);
+            });
```

Comment:
Its nice to see tests built up incrementally like this, however some of these tests are somewhat redundant. For example this first test is asserting that topics is an array, but in all of the following tests we are accessing and dealing with topics as if it is an array, which is indirectly confirming that topics is an array so you dont have to have a dedicated test for this.